### PR TITLE
NAS-114229 / 13.0 / add get_sectorsize_with_name to disk.pyx

### PR DESCRIPTION
**Pull request by yocalebo**
https://jira.ixsystems.com/browse/NAS-114229

### CODE REVIEW

File: bsd/disk.pyx [hunk 1]
```diff
@@ -5,6 +5,28 @@ from fcntl import ioctl
...
+
+
+def get_sectorsize_with_file(f):
+    """
+    Get sectorsize of disk.
+
+    `f` must be a python file object.
+    """
+    _buffer = array('B', range(0, 4))
+    ioctl(f.fileno(), disk_ioctl.DIOCGSECTORSIZE, _buffer, 1)
```
Comment (sonicaj) [inline bsd/disk.pyx:26]: Should we be checking it's return value to ensure it executed successfully ?

File: bsd/disk.pyx [hunk 2]
```diff
@@ -5,6 +5,28 @@ from fcntl import ioctl
...
 
 
+def get_sectorsize_with_name(disk):
+    """
+    Get sectorsize of disk.
+
+    `disk` must be a string and can be `/dev/da1` or just `da1`
+    """
+    disk = disk.removeprefix('/dev/')
+    with open(f'/dev/{disk}', 'rb') as f:
```
Comment (sonicaj) [inline bsd/disk.pyx:15]: Is this always guaranteed to exist ?

**Review by sonicaj — APPROVED** [review]

File: bsd/disk.pyx (hunk 1 above)
Comment (yocalebo) [reply bsd/disk.pyx:26]: This is _EXACTLY_ formatted the same way as the other methods. I don't feel like making this method different than the others.

File: bsd/disk.pyx (hunk 2 above)
Comment (yocalebo) [reply bsd/disk.pyx:15]: This is _EXACTLY_ formatted the same way as the other methods. I don't feel like making this method different than the others.

**Pull request merged**